ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerUtils.java

```diff
@@ -34,47 +35,36 @@
   private ServerUtils() {
   }
 
-  public static void handleError(final HttpServerResponse response, final int statusCode,
-      final int errorCode, final String errMsg) {
-    final ErrorResponse errorResponse = new ErrorResponse(errorCode, errMsg);
-    final Buffer buffer = errorResponse.toBuffer();
-    response.setStatusCode(statusCode).end(buffer);
-  }
-
-  public static void unhandledExceptionHandler(final Throwable t) {
-    log.error("Unhandled exception", t);
-  }
-
   public static <T> Optional<T> deserialiseObject(final Buffer buffer,
-      final HttpServerResponse response,
+      final RoutingContext routingContext,
       final Class<T> clazz) {
-    return PojoCodec.deserialiseObject(buffer, new HttpResponseErrorHandler(response), clazz);
+    return PojoCodec.deserialiseObject(buffer, new HttpResponseErrorHandler(routingContext), clazz);
   }
 
   private static class HttpResponseErrorHandler implements PojoDeserializerErrorHandler {
 
-    private final HttpServerResponse response;
+    private final RoutingContext routingContext;
 
-    HttpResponseErrorHandler(final HttpServerResponse response) {
-      this.response = response;
+    HttpResponseErrorHandler(final RoutingContext routingContext) {
+      this.routingContext = routingContext;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
ack